### PR TITLE
feat: breadcrumbs

### PR DESCRIPTION
**Pull request by Letushev**
Screenshot 2024-09-17 at 10 41 21
Screenshot 2024-09-17 at 10 41 27

Guide: https://remix.run/docs/en/main/guides/breadcrumbs.

- We don't show breadcrums on every page, only products category and product details.
- On product details we have a special case. Product details route is not nested inside category route. But we want to  show the category link in breadcrumbs if a user visits the product page directly from the category page. I do this using history state. So the category link appears only on the client side.

### CODE REVIEW

File: app/routes/product-details.$productSlug/route.tsx [hunk 1]
```diff
@@ -27,54 +31,91 @@ export const loader = async ({ params, request }: LoaderFunctionArgs) => {
...
+}
+
+export const handle: RouteHandle<typeof loader, ProductDetailsLocationState> = {
+    breadcrumb: (match, location) => {
+        const fromCategory = location.state?.fromCategory;
+
+        const productLink = (
+            <Link to={ROUTES.productDetails.to(match.data.product.slug!)} state={{ fromCategory }}>
+                {match.data.product.name!}
+            </Link>
```
Comment (tetianaman) [inline app/routes/product-details.$productSlug/route.tsx:48]: What do you think about `ProductLink` component?
We have also some product links on the home page. So user can use this component to sell specific product.

File: src/components/breadcrumbs/breadcrumbs.tsx [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { useLocation, useMatches } from '@remix-run/react';
+import { RouteMatch } from '~/router/types';
+import styles from './breadcrumbs.module.scss';
+import { ArrowRightIcon } from '../icons';
+
+export const Breadcrumbs = () => {
+    const matches = useMatches() as RouteMatch[];
+    const location = useLocation();
```
Comment (alisey) [inline src/components/breadcrumbs/breadcrumbs.tsx:8]: Will we be able to create a board for this component?

File: src/components/breadcrumbs/breadcrumbs.tsx [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+    return (
+        <div className={styles.breadcrumbs}>
+            {matches
+                .flatMap((match) => match.handle?.breadcrumb?.(match, location) ?? [])
+                .map((breadcrumb, index, arr) => [
+                    breadcrumb,
+                    index !== arr.length - 1 && <ArrowRightIcon className={styles.separatorIcon} />,
```
Comment (alisey) [inline src/components/breadcrumbs/breadcrumbs.tsx:16]: No warning about missing `key` on children?

**Review by alisey — APPROVED** [review]

File: src/components/breadcrumbs/breadcrumbs.tsx (hunk 2 above)
Comment (Letushev) [reply src/components/breadcrumbs/breadcrumbs.tsx:8]: Hope so. If not, I will create a dumb version.

File: src/components/breadcrumbs/breadcrumbs.tsx (hunk 3 above)
Comment (Letushev) [reply src/components/breadcrumbs/breadcrumbs.tsx]: No, which is weird. I've added React.Fragment with a key

**Pull request merged**